### PR TITLE
CI: run cri-o tests using overlay on ubuntu 17

### PR DESCRIPTION
**Pull request by chavafg**
Now that Ubuntu 17.10 has been added to the CI,
we can make use of this job to test CRI-O using overlay
storage driver.

Fixes: #204.

Signed-off-by: Salvador Fuentes <salvador.fuentes@intel.com>

### CODE REVIEW

File: integration/cri-o/cri-o.sh [hunk 1]
```diff
@@ -32,18 +33,24 @@ else
...
-
+# By default run CRI-O tests using devicemapper
+MAJOR=$(echo "$VERSION_ID"|cut -d\. -f1)
 export STORAGE_OPTIONS="--storage-driver devicemapper --storage-opt dm.use_deferred_removal=false"
+
+# But if on ubuntu 17.10 or newer, test using overlay
+# This will allow us to run test with at least 2 different
+# storage drivers.
+if [ "$ID" == "ubuntu" ] && [ "$MAJOR" -ge 17 ]; then
+	export CRIO_STORAGE_DRIVER_OPTS="--storage-driver=overlay"
```
Comment (sboeuf) [inline integration/cri-o/cri-o.sh]: nit: replace `--storage-driver=overlay` with `--storage-driver overlay` since that's the syntax used for devicemapper.

**Comment by sboeuf** [comment]
@chavafg Seems the CI is failing because of a wrong path for the rootfs. It seems unrelated to this PR.

File: integration/cri-o/cri-o.sh [hunk 2]
```diff
@@ -10,6 +10,7 @@ set -e
 SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
 source "${SCRIPT_PATH}/../../.ci/lib.sh"
 source "${SCRIPT_PATH}/crio_skip_tests.sh"
+source /etc/os-release
```
Comment (grahamwhaley) [inline integration/cri-o/cri-o.sh]: just a note - at some future point I suspect this will fail, as not all distros supply that file.

File: integration/cri-o/cri-o.sh [hunk 3]
```diff
@@ -32,18 +33,24 @@ else
...
-
+# By default run CRI-O tests using devicemapper
+MAJOR=$(echo "$VERSION_ID"|cut -d\. -f1)
 export STORAGE_OPTIONS="--storage-driver devicemapper --storage-opt dm.use_deferred_removal=false"
+
+# But if on ubuntu 17.10 or newer, test using overlay
+# This will allow us to run test with at least 2 different
+# storage drivers.
+if [ "$ID" == "ubuntu" ] && [ "$MAJOR" -ge 17 ]; then
+	export CRIO_STORAGE_DRIVER_OPTS="--storage-driver overlay"
```
Comment (grahamwhaley) [inline integration/cri-o/cri-o.sh]: I don't know how or where this trickles though - if it ends up as part of the docker config for instance or who config/supplies the backing store - I'm just going to trust this is the right thing and will work fine on how our CI 17.10 slave(s) are configured....

**Comment by grahamwhaley** [comment]
CI seems fine now - can somebody confirm this is all fixed/worked as we expected before we merge then?

File: integration/cri-o/cri-o.sh (hunk 2 above)
Comment (jodh-intel) [reply integration/cri-o/cri-o.sh]: Good point. It might be a good idea if we put that logic into `.ci/lib.sh` so that once you've sourced that file, you get access to `VERSION_ID`, etc.

File: integration/cri-o/cri-o.sh [hunk 4]
```diff
@@ -32,18 +33,24 @@ else
 	popd
 fi
 
-OLD_IFS=$IFS
```
Comment (devimc) [inline integration/cri-o/cri-o.sh]: why did you remove this?  I think you will have problems iterating over `skipCRIOTests[@]`

File: integration/cri-o/cri-o.sh [hunk 5]
```diff
@@ -32,18 +33,24 @@ else
...
 done
 
-IFS=$OLD_IFS
-
+# By default run CRI-O tests using devicemapper
+MAJOR=$(echo "$VERSION_ID"|cut -d\. -f1)
 export STORAGE_OPTIONS="--storage-driver devicemapper --storage-opt dm.use_deferred_removal=false"
+
+# But if on ubuntu 17.10 or newer, test using overlay
+# This will allow us to run test with at least 2 different
```
Comment (devimc) [inline integration/cri-o/cri-o.sh]: to run tests
